client/blocks/query-a8c-analytics-banners/index.tsx

```diff
@@ -0,0 +1,110 @@
+			__(
+				'These cookies allow us to optimize performance by collecting information on how users interact with our websites, including which pages are visited most, as well as other analytical data. We use these details to improve how our websites function and to understand how users interact with them. To learn more about how to control your cookie preferences, go to <a target="_blank" href="%1$s">%1$s</a>.'
+			),
+			'https://automattic.com/cookies/'
+		),
+		v2AdvertisingOptionHeading: __( 'Advertising' ),
+		v2AdvertisingOptionText: __(
+			'These cookies are set by us and our advertising partners to provide you with relevant content and to understand that content’s effectiveness. They may be used to collect information about your online activities over time and across different websites to predict your preferences and to display more relevant advertisements to you. These cookies also allow a profile to be built about you and your interests, and enable personalized ads to be shown to you based on your profile.'
+		),
+		v1ButtonText: __( 'Got It!' ),
+		v2CustomizeButtonText: __( 'Customize' ),
+		v2AcceptAllButtonText: __( 'Accept All' ),
+		v2AcceptSelectionButtonText: __( 'Accept Selection' ),
+	},
+};
+
+const onA8cAnalyticsLoaded = () => {
+	debug( '[ a8c-analytics.js ] Script has been loaded, Cookie Banner may be displayed' );
+};
+const onA8cTrackersFired = () => {
+	debug( '[ a8c-analytics.js ] Triggering allowed trackers in regard to user consent' );
+	loadTrackingScripts();
+};
+
+/**
+ * Component that loads a8c-analytics.js scripts into the React app functionality.
+ * That includes:
+ * - Cookie Banner
+ * - Do Not Sell dialog
+ *
+ * The script automatically identifies if user should see the banner, for reference see:
+ * https://opengrok.a8c.com/source/xref/wpcom/wp-content/mu-plugins/a8c-analytics/a8c-analytics.js?r=25b9de55
+ */
+const QueryAnalyticsBanners = () => {
+	useEffect(
+		() => {
+			// Assign config that will be used by a8c-analytics.js script
+			window.a8cAnalyticsConfig = a8cAnalyticsConfig;
+			// Add listener that will get triggered once a8c-analytics.js will be successfully loaded
+			document.addEventListener( 'a8c-analytics:loaded', onA8cAnalyticsLoaded );
+			document.addEventListener( 'a8c-analytics:fire-sensitive-pixels', onA8cTrackersFired );
+			// Load the script into the html document
+			loadScript(
+				`//s1.wp.com/wp-content/mu-plugins/a8c-analytics/a8c-analytics.js?v=${ A8C_ANALYTICS_CACHE_VERSION }`,
+				( error?: string ) => {
+					if ( error ) {
+						debug(
+							"[ a8c-analytics.js ] Couldn't load the script! Aborting displaying Cookie Banner"
+						);
+					}
+				}
+			);
+
+			return () => {
+				document.removeEventListener( 'a8c-analytics:loaded', onA8cAnalyticsLoaded );
+				document.removeEventListener( 'a8c-analytics:fire-sensitive-pixels', onA8cTrackersFired );
+			};
+		},
+		[
+			/* No dependencies, ensure that it run only once */
```

Comment:
I believe this comment is redundant, because an empty array as `useEffect` dependencies is a standard React idiom.